### PR TITLE
Move SQL 2019 extension's notebook code into Azure Data Studio

### PR DESCRIPTION
**Pull request by corivera**
Basic scenarios are passing, as well as the notebook unit tests. I still need to hook up the integration tests.

### CODE REVIEW

File: ThirdPartyNotices.txt [hunk 1]
```diff
@@ -12,15 +12,19 @@ expressly granted herein, whether by implication, estoppel or otherwise.
...
     angular2-slickgrid: https://github.com/Microsoft/angular2-slickgrid
     applicationinsights: https://github.com/Microsoft/ApplicationInsights-node.js
+    assert: https://github.com/browserify/commonjs-assert
     bootstrap: https://github.com/twbs/bootstrap
     chart.js: https://github.com/Timer/chartjs
     chokidar: https://github.com/paulmillr/chokidar
     comment-json: https://github.com/kaelzhang/node-comment-json
     core-js: https://github.com/zloirock/core-js
+    decompress: https://github.com/kevva/decompress
+    DefinitelyTyped: https://github.com/DefinitelyTyped/DefinitelyTyped
```
Comment (kevcunnane) [inline ThirdPartyNotices.txt]: Dev dependencies don't count 😄 we have to be shipping the things

File: extensions/notebook/package.json [hunk 2]
```diff
@@ -125,12 +218,69 @@
...
-		"vscode-nls": "^4.0.0"
+		"@jupyterlab/services": "^3.2.1",
+		"decompress": "^4.2.0",
+		"error-ex": "^1.3.1",
+		"figures": "^2.0.0",
+		"fs-extra": "^5.0.0",
+		"node-fetch": "^2.3.0",
+		"process-nextick-args": "^2.0.0",
+		"temp-write": "^3.4.0",
+		"vscode-languageclient": "^5.3.0-next.1",
```
Comment (kevcunnane) [inline extensions/notebook/package.json]: What do we use this for?

File: scripts/test-extensions.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -e
```
Comment (kevcunnane) [inline scripts/test-extensions.sh]: Please add tracking task on enabling this in our Azure DevOps validation builds (and code coverage).

File: extensions/notebook/src/prompts/list.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+// This code is originally from https://github.com/DonJayamanne/bowerVSCode
+// License: https://github.com/DonJayamanne/bowerVSCode/blob/master/LICENSE
+
+import { window } from 'vscode';
+import Prompt from './prompt';
+import EscapeException from './escapeException';
+
+export default class ListPrompt extends Prompt {
```
Comment (kevcunnane) [inline extensions/notebook/src/prompts/list.ts]: Let's create a task to clean up the use of the Prompt framework given how little we use it. We can basically delete it & replace easily.

File: scripts/test-extensions.sh (hunk 3 above)
Comment (corivera) [reply scripts/test-extensions.sh]: https://github.com/Microsoft/azuredatastudio/issues/4099

File: extensions/notebook/src/prompts/list.ts (hunk 4 above)
Comment (corivera) [reply extensions/notebook/src/prompts/list.ts]: https://github.com/Microsoft/azuredatastudio/issues/4100

**Pull request merged**